systemtest/src/main/java/io/strimzi/systemtest/resources/types/NamespaceType.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.resources.types;
+
+import io.fabric8.kubernetes.api.model.Namespace;
+import io.fabric8.kubernetes.api.model.NamespaceList;
+import io.fabric8.kubernetes.client.dsl.NonNamespaceOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.skodjob.testframe.interfaces.ResourceType;
+import io.skodjob.testframe.resources.KubeResourceManager;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaTopicUtils;
+import io.strimzi.systemtest.utils.kubeUtils.NamespaceUtils;
+
+import java.util.function.Consumer;
+
+public class NamespaceType implements ResourceType<Namespace> {
```

Comment:
I see that u utilized isDeleted, but cant the rest be removed and just extend the NamespaceType from TF?